kubernetes/cache/kube_cache.go

```diff
@@ -443,6 +443,69 @@ func (c *kubeCache) createKubernetesInformers(namespace string) informers.Shared
 	if namespace != "" {
 		opts = append(opts, informers.WithNamespace(namespace))
 	}
+	opts = append(
+		opts,
+		informers.WithTransform(func(obj interface{}) (interface{}, error) {
+			// similarly is trimmed in istio
+			t, ok := obj.(metav1.ObjectMetaAccessor)
+			if !ok {
+				// shouldn't happen
+				return obj, nil
+			}
+			// ManagedFields is large and we never use it
+			t.GetObjectMeta().SetManagedFields(nil)
```

Comment:
`t` is not actually used anywhere since down below different objects are returned so I don't think this does anything. If it's not easy to restructure this such that you can do `obj.GetObjectMeta().SetManagedFields(nil)` on the actual object that gets returned then I'd say don't bother with it. It save a few lines of code but it's basically being duplicated by returning a new `ObjectMeta` for the objects inside the case statements anyways.

Reply:
I think this is reference type, as similarly is done in Istio code:
https://github.com/istio/istio/blob/master/pkg/kube/util.go#L380
https://github.com/istio/istio/blob/master/pkg/kube/util.go#L355

Reply:
The difference is that they are always returning `obj` at the bottom whereas we create and return a new object inside the case statements, either `trimmedPod` or `trimmedService`.

Can you include some unit tests with the changes? You'll probably need to move this into its own function.